[Master] - Build fails due to missing tinyxml2 on openSUSE Tumbleweed and ArchLinux

## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Building the latest master fails on openSUSE Tumbleweed and ArchLinux due to missing tinyxml2, although the library is installed:

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Kodi should find tinyxml2.h and just build



## Actual Behavior
Building fails with error below



## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->



### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
Clone current master and start building with cmake:
%cmake \
    -DVERBOSE=ON \
    -DCMAKE_BUILD_TYPE=Release \
    -DCORE_SYSTEM_NAME=linux \
    -DAPP_RENDER_SYSTEM=gles \
    -DCORE_PLATFORM_NAME=gbm \
    -DGIT_VERSION="%{_git_version}" \
    -DENABLE_EVENTCLIENTS=ON \
    -DENABLE_INTERNAL_CROSSGUID=OFF \
    -DENABLE_INTERNAL_CDIO=OFF \
    -DENABLE_INTERNAL_FLATBUFFERS=OFF \
    -DENABLE_INTERNAL_FMT=OFF \
    -DENABLE_INTERNAL_FSTRCMP=OFF \
    -DENABLE_INTERNAL_RapidJSON=OFF \
    -DENABLE_INTERNAL_SPDLOG=OFF \
    -DENABLE_INTERNAL_DAV1D=OFF \
    -DENABLE_INTERNAL_UDFREAD=OFF \
    -DENABLE_INTERNAL_GTEST=OFF \
    -DENABLE_INTERNAL_KISSFFT=OFF \
    -DENABLE_INTERNAL_TINYXML2=OFF \
    -DENABLE_AIRTUNES=OFF \
    -DENABLE_DVDCSS=OFF \
    -DENABLE_PULSEAUDIO=OFF \
    -DENABLE_PIPEWIRE=OFF \
    -DENABLE_MYSQLCLIENT=OFF \
    -DENABLE_MARIADBCLIENT=ON \
    -DENABLE_GOLD=OFF \
    -DENABLE_MOLD=OFF \
    -DENABLE_INTERNAL_FFMPEG=ON \
    -DFFMPEG_URL="%{SOURCE9}" \
    -DENABLE_INTERNAL_CEC=OFF \
    -DENABLE_INTERNAL_NFS=OFF \
    -DENABLE_INTERNAL_PCRE=OFF \
    -DENABLE_INTERNAL_TAGLIB=OFF \
    -DENABLE_OPENGLES=ON \
    -DENABLE_VAAPI=ON \
    -DENABLE_VDPAU=OFF \
    -DUSE_LTO=OFF \
    -DBUILD_SHARED_LIBS=1 \
    -DENABLE_TESTING=OFF \
    -DENABLE_DEBUGFISSION=OFF


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
BuildLog Tumbleweed: (https://build.opensuse.org/package/live_build_log/home:Herbster0815:HTPC/kodi/openSUSE_Tumbleweed/x86_64)
Spec-file including "BuildRequires:  pkgconfig(tinyxml2)" : (https://build.opensuse.org/package/view_file/home:Herbster0815:HTPC/kodi/kodi.spec?expand=1)


BuildLog Arch:
-- Found TinyXML: /usr/lib/libtinyxml.so (found version "2.6.2")
-- MODULE: TINYXML2
-- LIB_TYPE: target
-- DEP_LOCATION: /home/mediapc/source/kodi-rpi-git-upstream/src/kodi-build/build
-- PROJECTSOURCE: /home/mediapc/source/kodi-rpi-git-upstream/src/xbmc-b9d83afe162307da8cd47ed9af416f536fa409bc
-- TINYXML2_URL: http://mirrors.kodi.tv/build-deps/sources/tinyxml2-9.0.0.tar.gz
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find TinyXML2 (missing: TINYXML2_LIBRARY) (found version "9.0.0")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindTinyXML2.cmake:107 (find_package_handle_standard_args)
  cmake/scripts/common/Macros.cmake:372 (find_package)
  cmake/scripts/common/Macros.cmake:386 (find_package_with_ver)
  CMakeLists.txt:226 (core_require_dep)


tinyxml2 is installed:

[11:15:42 @ Do Aug 10]root@Kleinerbaer[/home/mediapc/source/kodi-rpi-git-upstream/]$psearch tinyxml2
extra/tinyxml2 9.0.0-2 [Installiert]
    Simple XML parser made for easy integration
community/tinyxml2 9.0.0-1 [Installiert: 9.0.0-2]
    Simple XML parser made for easy integration

[11:15:47 @ Do Aug 10]root@Kleinerbaer[/home/mediapc/source/kodi-rpi-git-upstream/]$ls -l /usr/include/tinyxml2.h
-rw-r--r-- 1 root root 72K 18. Jul 15:02 /usr/include/tinyxml2.h



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: openSUSE Tumbleweed x64, ArchLinux armv7l
 - Kodi version: 21.x.+git.20230809T211603~b9d83afe1



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/cmake/modules/FindTinyXML2.cmake
+++ b/cmake/modules/FindTinyXML2.cmake
@@ -16,7 +16,7 @@ if(NOT TARGET tinyxml2::tinyxml2)
   SETUP_BUILD_VARS()
 
   # Check for existing TINYXML2. If version >= TINYXML2-VERSION file version, dont build
-  find_package(TINYXML2 CONFIG)
+  find_package(TINYXML2 CONFIG QUIET)
 
   # Some linux distro's dont package cmake config files for TinyXML2
   # This means that they will fall into the below and we will run a pkg_check_modules
@@ -57,36 +57,39 @@ if(NOT TARGET tinyxml2::tinyxml2)
       BUILD_DEP_TARGET()
     else()
       # This is the fallback case where linux distro's dont ship cmake config files
-      # use the old find_library way
+      # use the old find_library way. Only do this if we didnt find a cmake config 
+      # in the event of the version < depends version
+      if(NOT TARGET tinyxml2::tinyxml2)
+        if(PKG_CONFIG_FOUND)
+          pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
+        endif()
 
-      if(PKG_CONFIG_FOUND)
-        pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
-      endif()
-
-      find_path(TINYXML2_INCLUDE_DIR tinyxml2.h
-                                    PATHS ${PC_TINYXML2_INCLUDEDIR})
-      find_library(TINYXML2_LIBRARY_RELEASE NAMES tinyxml2
+        find_path(TINYXML2_INCLUDE_DIR tinyxml2.h
+                                      PATHS ${PC_TINYXML2_INCLUDEDIR})
+        find_library(TINYXML2_LIBRARY_RELEASE NAMES tinyxml2
+                                             PATHS ${PC_TINYXML2_LIBDIR})
+        find_library(TINYXML2_LIBRARY_DEBUG NAMES tinyxml2d
                                            PATHS ${PC_TINYXML2_LIBDIR})
-      find_library(TINYXML2_LIBRARY_DEBUG NAMES tinyxml2d
-                                         PATHS ${PC_TINYXML2_LIBDIR})
 
-      set(TINYXML2_VERSION ${PC_TINYXML2_VERSION})
-
+        set(TINYXML2_VERSION ${PC_TINYXML2_VERSION})
+      endif()
     endif()
 
-    add_library(tinyxml2::tinyxml2 UNKNOWN IMPORTED)
-    if(TINYXML2_LIBRARY_RELEASE)
+    if(NOT TARGET tinyxml2::tinyxml2)
+      add_library(tinyxml2::tinyxml2 UNKNOWN IMPORTED)
+      if(TINYXML2_LIBRARY_RELEASE)
+        set_target_properties(tinyxml2::tinyxml2 PROPERTIES
+                                                 IMPORTED_CONFIGURATIONS RELEASE
+                                                 IMPORTED_LOCATION_RELEASE "${TINYXML2_LIBRARY_RELEASE}")
+      endif()
+      if(TINYXML2_LIBRARY_DEBUG)
+        set_target_properties(tinyxml2::tinyxml2 PROPERTIES
+                                                 IMPORTED_CONFIGURATIONS DEBUG
+                                                 IMPORTED_LOCATION_DEBUG "${TINYXML2_LIBRARY_DEBUG}")
+      endif()
       set_target_properties(tinyxml2::tinyxml2 PROPERTIES
-                                               IMPORTED_CONFIGURATIONS RELEASE
-                                               IMPORTED_LOCATION_RELEASE "${TINYXML2_LIBRARY_RELEASE}")
+                                               INTERFACE_INCLUDE_DIRECTORIES "${TINYXML2_INCLUDE_DIR}")
     endif()
-    if(TINYXML2_LIBRARY_DEBUG)
-      set_target_properties(tinyxml2::tinyxml2 PROPERTIES
-                                               IMPORTED_CONFIGURATIONS DEBUG
-                                               IMPORTED_LOCATION_DEBUG "${TINYXML2_LIBRARY_DEBUG}")
-    endif()
-    set_target_properties(tinyxml2::tinyxml2 PROPERTIES
-                                             INTERFACE_INCLUDE_DIRECTORIES "${TINYXML2_INCLUDE_DIR}")
 
     if(TARGET tinyxml2)
       add_dependencies(tinyxml2::tinyxml2 tinyxml2)
@@ -94,9 +97,18 @@ if(NOT TARGET tinyxml2::tinyxml2)
   endif()
 
   if(TARGET tinyxml2::tinyxml2)
-    # Used to provide configure output of target information
-    get_target_property(TINYXML2_LIBRARY_DEBUG tinyxml2::tinyxml2 IMPORTED_LOCATION_DEBUG)
-    get_target_property(TINYXML2_LIBRARY_RELEASE tinyxml2::tinyxml2 IMPORTED_LOCATION_RELEASE)
+    get_target_property(_TINYXML2_CONFIGURATIONS tinyxml2::tinyxml2 IMPORTED_CONFIGURATIONS)
+    foreach(_tinyxml2_config IN LISTS _TINYXML2_CONFIGURATIONS)
+      # Some non standard config (eg None on Debian)
+      # Just set to RELEASE var so select_library_configurations can continue to work its magic
+      if((NOT ${_tinyxml2_config} STREQUAL "RELEASE") AND
+         (NOT ${_tinyxml2_config} STREQUAL "DEBUG"))
+        get_target_property(TINYXML2_LIBRARY_RELEASE tinyxml2::tinyxml2 IMPORTED_LOCATION_${_tinyxml2_config})
+      else()
+        get_target_property(TINYXML2_LIBRARY_${_tinyxml2_config} tinyxml2::tinyxml2 IMPORTED_LOCATION_${_tinyxml2_config})
+      endif()
+    endforeach()
+
     get_target_property(TINYXML2_INCLUDE_DIR tinyxml2::tinyxml2 INTERFACE_INCLUDE_DIRECTORIES)
   endif()
 

